### PR TITLE
NAS-104196 / 12.0 / Nas 104196b

### PR DESCRIPTION
**Pull request by dkmullen**
Adds new tables and pop-up forms for setting user/group quotas for datasets

### CODE REVIEW

File: src/app/helptext/storage/volumes/datasets/dataset-quotas.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+export default {
+    users: {
+        title: T('Dataset Users'),
+        action_label: T('Define Quotas for Selected Users'),
+        dialog: {
+          title: T('Define quotas for selected users'),
+          list: {
+            placeholder: T('Selected Users'),
+            tooltip: T('This list can be edited in the table. A quota change \
+ will apply to all users in this list.')
```
Comment (Mrt134) [inline src/app/helptext/storage/volumes/datasets/dataset-quotas.ts]:
```suggestion
            tooltip: T('This list can be edited in the table. A quota change \
 applies to all users in this list.')
```

File: src/app/helptext/storage/volumes/datasets/dataset-quotas.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+          },
+          data_quota: {
+            placeholder: T('User Data Quota ') + globalHelptext.human_readable.suggestion_label,
+            tooltip: T('Enter the amount of disk space that may be consumed by the selected users. \
```
Comment (Mrt134) [inline src/app/helptext/storage/volumes/datasets/dataset-quotas.ts]:
```suggestion
            tooltip: T('Amount of disk space that can be used by the selected users. \
```

File: src/app/helptext/storage/volumes/datasets/dataset-quotas.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+ Enter 0 (zero) to remove the quota.') + globalHelptext.human_readable.suggestion_tooltip
+          },
+          obj_quota: {
+            placeholder: T('User Object Quota'),
+            tooltip: T('Enter the number of objects which may be owned by each of the selected users. \
+Enter 0 (zero) to remove the quota.'),
```
Comment (Mrt134) [inline src/app/helptext/storage/volumes/datasets/dataset-quotas.ts]:
```suggestion
            tooltip: T('Number of objects that can be owned by each of the selected users. \
Entering <code>0</code> (zero) removes the quota.'),
```

File: src/app/helptext/storage/volumes/datasets/dataset-quotas.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+    groups: {
+      title: T('Dataset Groups'),
+      action_label: T('Define Quotas for Selected Groups'),
+      dialog: {
+        title: T('Define quotas for selected groups'),
+        list: {
+          placeholder: T('Selected Groups'),
+          tooltip: T('This list can be edited in the table. A quota change \
+will apply to all groups in this list.')
```
Comment (Mrt134) [inline src/app/helptext/storage/volumes/datasets/dataset-quotas.ts]:
```suggestion
applies to all groups in this list.')
```

File: src/app/helptext/storage/volumes/datasets/dataset-quotas.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        },
+        data_quota: {
+          placeholder: T('Group Data Quota ') + globalHelptext.human_readable.suggestion_label,
+          tooltip: T('Enter the amount of disk space that may be consumed by the selected groups. \
+Enter 0 (zero) to remove the quota.') + globalHelptext.human_readable.suggestion_tooltip
```
Comment (Mrt134) [inline src/app/helptext/storage/volumes/datasets/dataset-quotas.ts]:
```suggestion
          tooltip: T('Amount of disk space that can be consumed by the selected groups. \
Entering <code>0</code> (zero) removes the quota.') + globalHelptext.human_readable.suggestion_tooltip
```

File: src/app/helptext/storage/volumes/datasets/dataset-quotas.ts [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        },
+        obj_quota: {
+          placeholder: T('Group Object Quota'),
+          tooltip: T('Enter the number of objects which may be owned by each of the selected groups. \
+Enter 0 (zero) to remove the quota.'),
```
Comment (Mrt134) [inline src/app/helptext/storage/volumes/datasets/dataset-quotas.ts]:
```suggestion
          tooltip: T('Number of objects that can be owned by each of the selected groups. \
Entering <code>0</code> (zero) removes the quota.'),
```

**Comment by anodos325** [comment]
We should entries for results from the `pool.dataset.get_quota <dataset> <type>` where the quotas are greater than 0. When I was testing I could see all users (including ones without quotas).

**Comment by dkmullen** [comment]
The reason I'm showing all results is that I'm using the table not only to display data but also as the interface to chose users or groups on which to set quotas. ie, if the non-quotas were filtered out, you couldn't select them. You can also sort the table by quota (descending) which immediately puts all users/groups with quotas on top.

**Pull request closed without merging**